agent/api/ecsclient/client.go

```diff
@@ -226,7 +235,24 @@ func (client *APIECSClient) setInstanceIdentity(registerRequest ecs.RegisterCont
 	}
 
 	iidRetrieved := true
-	instanceIdentityDoc, err := client.ec2metadata.GetDynamicData(ec2.InstanceIdentityDocumentResource)
+	backoff := retry.NewExponentialBackoff(setInstIdRetryBackoffMin, setInstIdRetryBackoffMax,
+		setInstIdRetryBackoffJitter, setInstIdRetryBackoffMultiple)
+	ctx, cancel := context.WithTimeout(context.Background(), setInstIdRetryTimeOut)
+	defer cancel()
+	err := retry.RetryWithBackoffCtx(ctx, backoff, func() error {
+		var attemptErr error
+		seelog.Debugf("Attempting to get IID")
```

Comment:
Super nit: can we rephrase the log line as "Attempting to get instance identity document"